python/nav/ipdevpoll/jobs.py

```diff
@@ -244,9 +244,7 @@ def run(self):
         self._logger.debug("Job %r started with plugins: %r",
                            self.name, self.plugins)
         # Initialize netbox in container
-        nb = self._container_factory(shadows.Netbox, key=None,
-                                     id=self.netbox.id,
-                                     sysname=self.netbox.sysname)
+        self.containers.inject(None, self.netbox)
```

Comment:
I'm a bit wary of this.

If you inject the same instance into the Container Repository that you feed as an argument to plugins, plugins may end up changing the state/parameters of the current job before they are finished. The instance that gets fed to the Plugins  represents the object currently persisted in the database, and should ideally never mutate - whereas the instance in the container repository should represent information that was collected from the network during the job.

Reply:
this was the case for one of the plugins, and it has been updated. There were already numerous places where this.netbox were used rather than the container repository version